### PR TITLE
Add new exception hierarchy in class libraries.

### PR DESCRIPTION
**Pull request by iscottb122**

### CODE REVIEW

File: src/Expression/SymbolicaRuntimeException.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace Symbolica.Expression
+{
+    [Serializable]
+    public class SymbolicaRuntimeException : SymbolicaException
```
Comment (Choc13) [inline src/Expression/SymbolicaRuntimeException.cs:6]: Is this one supposed to be used for actual exceptions? I'm not sure the name is too obvious here as all exceptions are runtime exceptions aren't they?

File: src/Implementation/Exceptions/UnsupportedArchitectureException.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using Symbolica.Expression;
+
+namespace Symbolica.Implementation.Exceptions
+{
+    [Serializable]
+    public class UnsupportedArchitectureException : SymbolicaException
```
Comment (Choc13) [inline src/Implementation/Exceptions/UnsupportedArchitectureException.cs:7]: Is this not one we want to show to users?

File: src/Expression/SymbolicaRuntimeException.cs (hunk 1 above)
Comment (iscottb122) [reply src/Expression/SymbolicaRuntimeException.cs]: As discussed, it's gone.

File: src/Implementation/Exceptions/UnsupportedArchitectureException.cs (hunk 2 above)
Comment (iscottb122) [reply src/Implementation/Exceptions/UnsupportedArchitectureException.cs]: Yes, and it was, but now more explicit since it is an `ErrorException`.

**Review by Choc13 — APPROVED** [review]

**Pull request merged**